opendbc/car/byd/carstate.py

```diff
@@ -0,0 +1,145 @@
+from opendbc.car import Bus, structs
+from opendbc.can.parser import CANParser
+from opendbc.can.can_define import CANDefine
+from opendbc.car.common.conversions import Conversions as CV
+from opendbc.car.interfaces import CarStateBase
+from opendbc.car.byd.values import DBC, CANBUS
+
+# Multiplier between GPS ground speed to the meter cluster's displayed speed
+HUD_MULTIPLIER = 1.068
+
+class CarState(CarStateBase):
+  def __init__(self, CP):
+    super().__init__(CP)
+    print(CP.carFingerprint)
+    can_define = CANDefine(DBC[CP.carFingerprint][Bus.pt])
+
+    self.shifter_values = can_define.dv["DRIVE_STATE"]['GEAR']
+    self.set_distance_values = can_define.dv['ACC_HUD_ADAS']['SET_DISTANCE']
+
+    self.prev_angle = 0
+    self.lss_state = 0
+    self.lss_alert = 0
+    self.tsr = 0
+    self.ahb = 0
+    self.passthrough = 0
+    self.lka_on = 0
+    self.HMA = 0
+    self.lkas_rdy_btn = False
+    self.lkas_faulted = False
+
+  def update(self, can_parsers) -> structs.CarState:
+    cp = can_parsers[Bus.pt]
+    cp_cam = can_parsers[Bus.cam]
+    ret = structs.CarState()
+
+    self.tsr = cp_cam.vl["LKAS_HUD_ADAS"]['TSR']
+    self.lka_on = cp_cam.vl["LKAS_HUD_ADAS"]['STEER_ACTIVE_ACTIVE_LOW']
+    self.lkas_rdy_btn = cp.vl["PCM_BUTTONS"]['LKAS_ON_BTN']
+    self.abh = cp_cam.vl["LKAS_HUD_ADAS"]['SET_ME_XFF']
+    self.passthrough = cp_cam.vl["LKAS_HUD_ADAS"]['SET_ME_X5F']
+    self.HMA = cp_cam.vl["LKAS_HUD_ADAS"]['HMA']
+    self.lkas_healthy = cp_cam.vl["STEERING_MODULE_ADAS"]['EPS_OK']
+
+    # EV irrelevant messages
+    ret.brakeHoldActive = False
+
+    ret.wheelSpeeds = self.get_wheel_speeds(
+      cp.vl["WHEEL_SPEED"]['WHEELSPEED_FL'],
+      cp.vl["WHEEL_SPEED"]['WHEELSPEED_FL'],
+      cp.vl["WHEEL_SPEED"]['WHEELSPEED_BL'],
+      cp.vl["WHEEL_SPEED"]['WHEELSPEED_BL'],
+    )
+    ret.vEgoRaw = (ret.wheelSpeeds.rl + ret.wheelSpeeds.fl) / 2.0
+
+    # unfiltered speed from CAN sensors
+    ret.vEgo, ret.aEgo = self.update_speed_kf(ret.vEgoRaw)
+    ret.vEgoCluster = ret.vEgo
+    ret.standstill = ret.vEgoRaw < 0.01
+
+    # safety checks to engage
+    can_gear = int(cp.vl["DRIVE_STATE"]['GEAR'])
+
+    ret.doorOpen = any([cp.vl["METER_CLUSTER"]['BACK_LEFT_DOOR'],
+                     cp.vl["METER_CLUSTER"]['FRONT_LEFT_DOOR'],
+                     cp.vl["METER_CLUSTER"]['BACK_RIGHT_DOOR'],
+                     cp.vl["METER_CLUSTER"]['FRONT_RIGHT_DOOR']])
+
+    ret.seatbeltUnlatched = cp.vl["METER_CLUSTER"]['SEATBELT_DRIVER'] == 0
+    ret.gearShifter = self.parse_gear_shifter(self.shifter_values.get(can_gear, None))
+
+    # gas pedal
+    ret.gas = cp.vl["PEDAL"]['GAS_PEDAL']
+    ret.gasPressed = ret.gas >= 0.01
+
+    # brake pedal
+    ret.brake = cp.vl["PEDAL"]['BRAKE_PEDAL']
+    ret.brakePressed = bool(cp.vl["DRIVE_STATE"]["BRAKE_PRESSED"]) or ret.brake > 0.01
```

Comment:
We fell into the trap of using two signals for brake pressed on GM which lead to many bugs. Can we just use one? Why not?